game/action.go

```diff
@@ -12,16 +12,16 @@ type Action struct {
 	AfterEffect *Effect
 	Card        *Card
 	// the spell target Card's coming into play effect
-	EntersTheBattleFieldSpellTarget *StackObject
-	Cost                            *Cost
+	EntersTheBattleFieldSpellTargetId StackObjectId
```

Comment:
Just don't use the trailing "Id" on the names of things - the permanents aren't like that, and it's sort of equivalent to naming the pointers FooPointer rather than just Foo. Ie the name of this can just be EntersTheBattleFieldSpellTarget.